proposals/shared-everything-threads/Overview.md

```diff
@@ -234,33 +370,291 @@ fingerprinting. Note that:
+   shared structs][proposal-structs] proposal.
+ - Shared JS structs, shared JS arrays, `Atomics.Mutex` and any other shareable type introduced by
+   the JS shared structs proposal can be converted to shared externref.
+
+> Note: It would be great for Wasm-JS interoperability if JS shared structs and arrays could also be
+> converted to Wasm shared structs and arrays and vice versa. We will have to see what is possible
+> there.
+
+Conversely, `ToJSValue(w)` should convert WebAssembly shared references to shareable opaque objects
+on the JS side, analogous to how it converts non-shared references to opaque JS objects, at least
+until we get more ergonomic interop between JS and WasmGC. These opaque JS objects will be shared
+objects that do not require rewrapping, and their identities survive round trips via postMessage
+across thread boundaries.
+
+Waitqueue references are converted to `WebAssembly.WaitQueue` objects, described below.
+
+[proposal-structs]: https://github.com/tc39/proposal-structs
+
+#### Shared annotations
+
+Just like the original threads proposal exposed shared memories in the JS API, this proposal exposes
+shared functions, tables, globals, tags, and exceptions. Unlike shared `WebAssembly.Memory` objects,
+which are actually realm-local wrappers around SharedArrayBuffers (which are in turn realm-local
+wrappers around the underlying shared buffers), the new shared tables, globals, tags, and exceptions
+will be shared JS objects that do not require rewrapping. Such shared objects have fixed layout and
+behave mostly like [sealed] JS objects, and their identities survive round trips via postMessage
+across thread boundaries. Additionally, to facilitate calling methods on these shared objects, they
+have [realm-local prototypes][tls-prototypes].
+
+The shared versions of the JS API types (besides memories) will be represented with new
+constructors:
+
+ - `WebAssembly.SharedFunction`
+ - `WebAssembly.SharedTable`
+ - `WebAssembly.SharedGlobal`
+ - `WebAssembly.SharedTag`
+ - `WebAssembly.SharedException`
+
+These new types provide an API that mirrors that provided by their unshared analogues. Like shared
+memories, shared tables must be initialized with a maximum size. Shared exceptions may only be
+created with shared tags. Storing unshared data into a shared global or table throws an exception.
+
+While the behaviors chosen here mirror those proposed in the JS shared structs proposal,
+implementation of this behavior does not require any changes in the JS specification because this
+proposal does not propose any user-programmable means in JS to create new kinds of shared objects.
+
+We also expose realm-local globals with a `realm-local` option in the `WebAssembly.SharedGlobal`
+constructor.
+
+> Note: It would also be possible to reuse the existing constructors and add a `shared` property for
+> testing rather than use new constructors, which allow testing with `instanceof`. This would be
```

Comment:
Yeah, I doubt instanceof is of much use in this context. If you don't know a type of a thing from context and instead have to handle it generically, then you'll usually have to inspect the type via reflection anyway.